rnote-ui/src/canvas/mod.rs

```diff
@@ -571,6 +602,73 @@ impl RnoteCanvas {
+                    match file.monitor_file(gio::FileMonitorFlags::WATCH_MOVES, gio::Cancellable::NONE) {
+                        Ok(output_file_monitor) => {
+                            let output_file_monitor_signal = output_file_monitor.connect_changed(
+                                glib::clone!(@weak canvas, @weak appwindow => move |_monitor, file, other_file, event| {
+                                    let dispatch_toast_reload_modified_file = || {
+                                        appwindow.canvas().set_unsaved_changes(true);
+
+                                        appwindow.dispatch_toast_w_button_singleton(&gettext("Opened file was modified on disk."), &gettext("Reload"), clone!(@weak appwindow => move |_reload_toast| {
+                                            if let Some(output_file) = appwindow.canvas().output_file() {
+                                                if let Err(e) = appwindow.load_in_file(&output_file, None) {
+                                                    log::error!("failed to reload current output file, {}", e);
+                                                }
+                                            }
+                                        }), 0, &mut canvas.imp().output_file_modified_toast_singleton.borrow_mut());
+                                    };
+
+                                    match event {
+                                        gio::FileMonitorEvent::Changed => {
+                                            dispatch_toast_reload_modified_file();
+                                        },
+                                        gio::FileMonitorEvent::Renamed => {
+                                            // if previous file name was .goutputstream-<hash>, then the file has been replaced using gio (most likely by another rnote instance).
+                                            // => file has been modified, handle it the same as the Changed event.
+                                            if let Some(old_path) = file.path() {
+                                                if let Some(old_file_name) = old_path.file_name() {
+                                                    if String::from(old_file_name.to_string_lossy()).starts_with(".goutputstream-") {
+                                                        dispatch_toast_reload_modified_file();
+                                                        return;
+                                                    }
+                                                }
+                                            }
+
+                                            // other_file *should* never be none.
+                                            if other_file.is_none() {
+                                                canvas.set_unsaved_changes(true);
+                                            }
+
+                                            canvas.set_output_file(other_file.cloned());
```

Comment:
I would dispatch a "regular" toast here, just to let the user know that the file got renamed.